servers/curated-corpus-api/src/admin/resolvers/mutations/PublisherDomain/createOrUpdatePublisherDomain.integration.ts

```diff
@@ -0,0 +1,268 @@
+  const headers = {
+    name: 'Test User',
+    username: 'test.user@test.com',
+    groups: `group1,group2,${MozillaAccessGroup.SCHEDULED_SURFACE_CURATOR_FULL}`,
+  };
+
+  // Headers for a user WITHOUT corpus write access
+  const headersNoAccess = {
+    name: 'No Access User',
+    username: 'no.access@test.com',
+    groups: 'group1,group2',
+  };
+
+  beforeAll(async () => {
+    // port 0 tells express to dynamically assign an available port
+    ({ app, adminServer: server, adminUrl: graphQLUrl } = await startServer(0));
+    db = client();
+    await clearDb(db);
+  });
+
+  afterAll(async () => {
+    await server.stop();
+    await db.$disconnect();
+  });
+
+  beforeEach(async () => {
+    await clearDb(db);
+  });
+
+  describe('successful operations', () => {
+    it('should create a new PublisherDomain mapping', async () => {
+      const input = {
+        domainName: 'example.com',
+        publisher: 'Example Publisher',
+      };
+
+      const result = await request(app)
+        .post(graphQLUrl)
+        .set(headers)
+        .send({
+          query: print(CREATE_OR_UPDATE_PUBLISHER_DOMAIN),
+          variables: { data: input },
+        });
+
+      expect(result.body.errors).toBeUndefined();
+      expect(result.body.data).not.toBeNull();
+
+      const response = result.body.data?.createOrUpdatePublisherDomain;
+      expect(response.domainName).toEqual('example.com');
+      expect(response.publisher).toEqual('Example Publisher');
+      expect(response.createdBy).toEqual('test.user@test.com');
+      expect(response.createdAt).toBeDefined();
+      expect(response.updatedAt).toBeDefined();
+      expect(response.updatedBy).toBeNull();
+    });
+
+    it('should update an existing PublisherDomain mapping', async () => {
+      // First create a record
+      await db.publisherDomain.create({
+        data: {
+          domainName: 'example.com',
+          publisher: 'Old Publisher Name',
+          createdBy: 'original.user@test.com',
+        },
+      });
+
+      const input = {
+        domainName: 'example.com',
+        publisher: 'New Publisher Name',
+      };
+
+      const result = await request(app)
+        .post(graphQLUrl)
+        .set(headers)
+        .send({
+          query: print(CREATE_OR_UPDATE_PUBLISHER_DOMAIN),
+          variables: { data: input },
+        });
+
+      expect(result.body.errors).toBeUndefined();
+      expect(result.body.data).not.toBeNull();
+
+      const response = result.body.data?.createOrUpdatePublisherDomain;
+      expect(response.domainName).toEqual('example.com');
+      expect(response.publisher).toEqual('New Publisher Name');
+      // Original creator should be preserved
+      expect(response.createdBy).toEqual('original.user@test.com');
+      // updatedBy should be the current user
+      expect(response.updatedBy).toEqual('test.user@test.com');
+    });
+
+    it('should normalize domain name (lowercase, strip www)', async () => {
```

Comment:
for the same reason you give below (functionality is tested already in unit tests), i don't really think we need these domain formatting/validation checks as integration tests.

if you want to verify that the resolver is actually calling those normalization/validation functions, you could do that with mocks in a unit test.